phoenicis-library/src/main/java/org/phoenicis/library/ShortcutManager.java

```diff
@@ -162,28 +163,34 @@ public void updateShortcut(ShortcutDTO shortcutDTO) {
         // backup icon if it didn't change (deleteShortcut will delete it -> icon lost after shortcut update)
         final File iconFile = new File(shortcutDirectory, baseName + ".icon");
         final File iconBackup = new File(shortcutDirectory, baseName + ".icon_backup");
-
-        final boolean keepIcon = shortcutDTO.getIcon().getPath().equals(iconFile.getPath());
-        if (keepIcon) {
-            try {
-                Files.move(iconFile.toPath(), iconBackup.toPath());
-                shortcutDTO = new ShortcutDTO.Builder(shortcutDTO).withIcon(iconBackup.toURI()).build();
-            } catch (IOException e) {
-                LOGGER.error("Could not backup icon.");
+        final URI shortcutIcon = shortcutDTO.getIcon();
+
+        if (shortcutIcon != null) {
+            final boolean keepIcon = shortcutIcon.getPath().equals(iconFile.getPath());
+            if (keepIcon) {
+                try {
+                    Files.move(iconFile.toPath(), iconBackup.toPath());
+                    shortcutDTO = new ShortcutDTO.Builder(shortcutDTO).withIcon(iconBackup.toURI()).build();
+                } catch (IOException e) {
+                    LOGGER.error("Could not backup icon.");
+                }
             }
         }
 
         // backup miniature if it didn't change (deleteShortcut will delete it -> miniature lost after shortcut update)
         final File miniatureFile = new File(shortcutDirectory, baseName + ".miniature");
         final File miniatureBackup = new File(shortcutDirectory, baseName + ".miniature_backup");
-
-        final boolean keepMiniature = shortcutDTO.getMiniature().getPath().equals(miniatureFile.getPath());
-        if (keepMiniature) {
-            try {
-                Files.move(miniatureFile.toPath(), miniatureBackup.toPath());
-                shortcutDTO = new ShortcutDTO.Builder(shortcutDTO).withMiniature(miniatureBackup.toURI()).build();
-            } catch (IOException e) {
-                LOGGER.error("Could not backup miniature.");
+        final URI shortcutMiniature = shortcutDTO.getMiniature();
+
+        if (shortcutMiniature != null) {
+            final boolean keepMiniature = shortcutMiniature.getPath().equals(miniatureFile.getPath());
+            if (keepMiniature) {
```

Comment:
I kind of think an `Optional` approach would be shorter (doesn't need so many indentions):

```java
Optional.ofNullable(shortcutDTO.getMiniature())
    .filter(shortcutMiniature -> shortcutMiniature.getPath().equals(miniatureFile.getPath()))
    .ifPresent(shortcutMiniature -> ...);
```

Reply:
If we want an `Optional`, I think we should do it properly and use `Optional` in the DTO already. However, I really wouldn't want to do this now/in this PR (not sure which effects it will have in the scripts repo etc.).